tests/instances.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { test, Page } from "@playwright/test";
+import { TEXT_TIMEOUT } from "./constants";
+
+test("instance create and remove", async ({ page }) => {
+  const instance = randomInstanceName();
+  await createInstance(page, instance);
+  await deleteInstance(page, instance);
+});
+
+test("snapshot create restore and remove ", async ({ page }) => {
+  const instance = randomInstanceName();
+  const snapshot = randomSnapshotName();
+  await createInstance(page, instance);
+  await openInstanceDetail(page, instance);
+  await createSnapshot(page, instance, snapshot);
+  await page.waitForSelector(
+    `text=Snapshot ${snapshot} created.`,
+    TEXT_TIMEOUT
+  );
+
+  await restoreSnapshot(page, snapshot);
+  await page.waitForSelector("text=Snapshot restored", TEXT_TIMEOUT);
+
+  await deleteSnapshot(page, snapshot);
+  await page.waitForSelector("text=Snapshot deleted", TEXT_TIMEOUT);
+
+  await deleteInstance(page, instance);
+});
+
+const randomInstanceName = (): string => {
+  const r = (Math.random() + 1).toString(36).substring(7);
+  return `playwright-instance-${r}`;
+};
+
+const randomSnapshotName = (): string => {
+  const r = (Math.random() + 1).toString(36).substring(7);
+  return `playwright-snapshot-${r}`;
+};
+
+const createInstance = async (page: Page, instanceName: string) => {
+  await page.goto("/ui/");
+  await page.getByRole("link", { name: "Instances" }).click();
+  await page.getByRole("button", { name: "Add instance" }).click();
+  await page.getByLabel("Instance name").click();
+  await page.getByLabel("Instance name").fill(instanceName);
+  await page.getByRole("button", { name: "Select image" }).click();
+  await page
+    .getByPlaceholder("Search for distributions, releases or or aliases")
+    .click();
+  await page
+    .getByPlaceholder("Search for distributions, releases or or aliases")
+    .fill("alpine");
+  await page
+    .getByRole("row", {
+      name: "Distribution Release Architecture Alias Action",
+    })
+    .getByRole("button", { name: "Select" })
+    .first()
+    .click();
+  await page.getByRole("button", { name: "Create" }).first().click();
+  await page
+    .getByRole("paragraph")
+    .getByRole("link", { name: instanceName })
+    .click();
+};
+
+const deleteInstance = async (page: Page, instanceName: string) => {
+  await openInstanceDetail(page, instanceName);
+  await page.waitForTimeout(10000).then(async () => {
+    await page.keyboard.down("Shift");
+    await page.getByRole("button", { name: "Delete Delete" }).click();
+    await page.keyboard.up("Shift");
```

Comment:
Unlike all the other confirmation cases, here we are confirming the action with the `Shift` shortcut. Should the function be called `deleteInstanceNoConfirm` to make it clear?